### ISSUE TITLE
Ternary vs. Nullable parsing issue

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#6957]
This was reported to us by @VladimirReshetnikov in 2009.

> I believe this code should compile without errors:
```cs
class P
{
    static void Main()
    {
        object x = null as int? & 1;
    }
}
```

**Comment by leppie** [comment]
Along the same lines this fails on the legacy compiler (to test Roslyn):
```
object a = (int?)1;
object w = a is int? & true;
```

**Comment by gafter** [comment]
I do not there is ever a reason to write `is Type?` in source; you can always accomplish the same thing writing `is Type`. I therefore don't think this issue is urgent.

**Comment by leppie** [comment]
@gafter I agree it is nonsensical, but it should still parse correctly, no?

**Comment by gafter** [comment]
@leppie It is hard to say how much effort we should invest in unbounded lookahead in the parser for such nonsense. I'l actually rather forbid it in the language spec but have the compiler accept some cases for compatibility.

**Comment by leppie** [comment]
@gafter: If I had my way, nullable valuetypes would not exist past some user-implementable `Nullable<T>` form :)
